javax/src/main/java/org/jboss/ejb/protocol/remote/EJBServerChannel.java

```diff
@@ -134,6 +135,11 @@ final class EJBServerChannel {
             configuration.setObjectResolver(new ProtocolV1ObjectResolver(channel.getConnection(), true));
             configuration.setVersion(2);
         } else {
+            if (version < Protocol.JAKARTAEE_PROTOCOL_VERSION && Protocol.LATEST_VERSION >= Protocol.JAKARTAEE_PROTOCOL_VERSION) {
+                // EJB server uses EJB PROTOCOL version 4 or above but EJB client uses EJB PROTOCOL version 3 or below
+                // so in this case we need to translate classes from JavaEE API to JakartaEE API and vice versa
+                configuration.setClassNameTransformer(ClassNameTransformer.JAVAEE_TO_JAKARTAEE);
+            }
```

Comment:
From the above comment, when one end is using protocol 3 or below and the latest supported version is 4+, the transformation will occur. But looking at the if clause above, when `version` is 2, the transformer is not registered. Should this block be restructured?

Reply:
Of course you are right @chengfang . Good catch!